Mathlib/Analysis/LocallyConvex/Basic.lean

```diff
@@ -278,6 +281,19 @@ protected theorem Balanced.convexHull (hs : Balanced 𝕜 s) : Balanced 𝕜 (co
   simp only [smul_add, ← smul_comm]
   exact convex_convexHull ℝ s (hx a ha) (hy a ha) hu hv huv
 
+variable (F ℱ : Type*) [AddCommGroup F] [Module 𝕜 F]
+variable [FunLike ℱ F E] [LinearMapClass ℱ 𝕜 F E]
+
+theorem Absorbent.subset_range_iff_surjective {f : ℱ} {s : Set E} (hs_abs : Absorbent 𝕜 s) :
+    s ⊆ Set.range f ↔ (⇑f).Surjective := by
+  refine ⟨fun hs_sub y ↦ ?_, by simp_all⟩
+  obtain ⟨r, -, hr⟩ := Absorbs.exists_pos (hs_abs y)
+  obtain ⟨x, hx⟩ := NormedField.exists_lt_norm 𝕜 r
+  specialize hr _ hx.le
+  grw [hs_sub, Set.singleton_subset_iff] at hr
+  obtain ⟨-, ⟨z, rfl⟩, rfl⟩ := hr
+  exact ⟨x • z, map_smul _ _ _⟩
```

Comment:
What would you think about splitting this into
- a lemma saying that an absorbent vector subspace is everything
- then this follows using [Absorbent.mono](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Bornology/Absorbs.html#Absorbent.mono)

For example, I think this line of proof allows generalizing to semilinear maps (under the hypotheses of [LinearMap.range](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Submodule/Range.html#LinearMap.range))

Reply:
Nice idea! Let me put the PR on WIP while I'll implement this, I'll ping you once it is done.